tests/test_plugin_random_seed.py

```diff
@@ -0,0 +1,139 @@
+        cls.run_number = "TestRun_00000"
+
+        cls.test_context = fuse.context.full_chain_context(
+            cls.temp_dir.name, run_without_proper_corrections=True
+        )
+        cls.test_context.set_config(
+            {
+                "path": cls.temp_dir.name,
+                "file_name": test_root_file_name,
+                "entry_start": 0,
+                "entry_stop": 10,
+            }
+        )
+
+        # Get all registered fuse plutins.
+        cls.all_registered_fuse_plugins = {}
+        for key, value in cls.test_context._plugin_class_registry.items():
+            if "fuse" in str(value):
+                cls.all_registered_fuse_plugins[key] = value
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.temp_dir.cleanup()
+
+    def setUp(self):
+        downloader = straxen.MongoDownloader(store_files_at=(self.temp_dir.name,))
+        downloader.download_single(test_root_file_name, human_readable_file_name=True)
+
+        assert os.path.exists(os.path.join(self.temp_dir.name, test_root_file_name))
+
+    def tearDown(self):
+        # self.temp_dir.cleanup()
+        shutil.rmtree(self.temp_dir.name)
+        os.makedirs(self.temp_dir.name)
+
+    @timeout_decorator.timeout(TIMEOUT, exception_message="test_if_plugins_get_user_seed timed out")
+    def test_if_plugins_get_user_seed(self):
+        self.test_context.set_config(
+            {
+                "deterministic_seed": False,
+                "user_defined_random_seed": 42,
+            }
+        )
+
+        # Lets check the random seed for all fuse plugins
+        for key in self.all_registered_fuse_plugins.keys():
+            plugin = self.test_context.get_single_plugin(self.run_number, key)
+
+            if hasattr(plugin, "seed"):
+                assert (
+                    plugin.seed == 42
+                ), f"Expecting seed to be 42, but got {plugin.seed} for {key} plugin"
+
+    @timeout_decorator.timeout(
+        TIMEOUT, exception_message="test_if_plugins_with_rng_have_a_proper_seed timed out"
+    )
+    def test_if_plugins_with_rng_have_a_proper_seed(self):
+
+        # Lets check the random seed for all fuse plugins
+        for key in self.all_registered_fuse_plugins.keys():
+            plugin = self.test_context.get_single_plugin(self.run_number, key)
+
+            if hasattr(plugin, "rng"):
+                if not hasattr(plugin, "seed"):
+                    raise ValueError(f"Plugin {key} has rng but no seed")
+
+    @timeout_decorator.timeout(
+        TIMEOUT, exception_message="test_if_negative_seeds_are_intercepted timed out"
+    )
+    def test_if_negative_seeds_are_intercepted(self):
+        self.test_context.set_config(
+            {
+                "deterministic_seed": False,
+                "user_defined_random_seed": -42,
+            }
+        )
+
+        # Lets check the random seed for all fuse plugins
+        for key in self.all_registered_fuse_plugins.keys():
+
+            try:
```

Comment:
Please use `with self.assertRaises(AssertionError):` but not barely `try except`.

Reply:
Ahh nice! I did not know this feature yet! For plugins that have no seed I need to raise the AssertionError manually so the test will not fail.